zone/client_packet.cpp

```diff
@@ -16533,3 +16644,306 @@ void Client::ReloadExpansionProfileSetting()
+	LogTribute("Received OP_Handle_OP_GuildTributeDonateItem of length [{}]", app->size);
+
+	if (app->size != sizeof(GuildTributeDonateItemRequest_Struct)) {
+		LogError("Invalid size on Handle_OP_GuildTributeDonateItemRequest packet");
+		return;
+	}
+
+	auto in = (GuildTributeDonateItemRequest_Struct*)app->pBuffer;
+
+	const auto* inst = GetInv().GetItem(in->slot);
+	auto favor = inst->GetItemGuildFavor() * in->quantity;
+
+	auto guild = guild_mgr.GetGuildByGuildID(guild_id);
+	if (guild) {
+		guild->tribute.favor += favor;
+		guild_mgr.UpdateDbGuildFavor(GuildID(), guild->tribute.favor);
+		auto member_favor = guild_mgr.UpdateDbMemberFavor(GuildID(), CharacterID(), favor);
+
+		if (inst->IsStackable()) {
+			if (inst->GetCharges() < (int32)in->quantity) {
+				favor = 0;
+			}
+			DeleteItemInInventory(in->slot, in->quantity, false, true);
+		}
+		else {
+			DeleteItemInInventory(in->slot, 0, false, true);
+		}
+
+		SendGuildTributeDonateItemReply(in, favor);
+
+		if (RuleB(QueryServ, PlayerLogGuildTributeItemDonations)) {
+			auto event_desc = fmt::format("Guild Item Donation :: {} donated item id {} for {} favor in zone {}\.",
+				GetName(),
+				inst->GetID(),
+				favor,
+				GetZoneID()
+			);
+			QServ->PlayerLogEvent(Player_Log_Guild_Tribute_Item_Donation, CharacterID(), event_desc);
+		}
+
+		auto outapp = new ServerPacket(ServerOP_GuildTributeUpdateDonations, sizeof(GuildTributeUpdate));
+		GuildTributeUpdate* out = (GuildTributeUpdate*)outapp->pBuffer;
+		out->guild_id = GuildID();
+		out->favor = guild->tribute.favor;
+		strncpy(out->player_name, GetCleanName(), strlen(GetCleanName()));
+		out->member_time = time(nullptr);
+		out->member_enabled = GuildTributeOptIn();
+		out->member_favor = member_favor;
+		worldserver.SendPacket(outapp);
+		safe_delete(outapp);
+	}
+}
+
+void Client::Handle_OP_GuildTributeDonatePlat(const EQApplicationPacket* app)
+{
+	LogTribute("Received OP_Handle_OP_GuildTributeToggle of length [{}]", app->size);
+
+	if (app->size != sizeof(GuildTributeDonatePlatRequest_Struct)) {
+		LogError("Invalid size on Handle_OP_GuildTributeDonatePlatRequest_Struct packet");
+		return;
+	}
+
+	auto in = (GuildTributeDonatePlatRequest_Struct*)app->pBuffer;
+
+	auto quantity = in->quantity;
+
+	auto favor = quantity * RuleI(Guild, TributePlatConversionRate);
+	auto guild = guild_mgr.GetGuildByGuildID(guild_id);
+	if (guild) {
+		guild->tribute.favor += favor;
+		guild_mgr.UpdateDbGuildFavor(GuildID(), guild->tribute.favor);
+		auto member_favor = guild_mgr.UpdateDbMemberFavor(GuildID(), CharacterID(), favor);
+
+		TakePlatinum(quantity, false);
+		SendGuildTributeDonatePlatReply(in, favor);
+
+		if (RuleB(QueryServ, PlayerLogGuildTributePlatDonations)) {
+			auto event_desc = fmt::format("Guild Platinum Donation :: {} donated {} platinum for {} favor in zone {}\.",
+				GetName(),
+				quantity,
+				favor,
+				GetZoneID()
+			);
+			QServ->PlayerLogEvent(Player_Log_Guild_Tribute_Plat_Donation, CharacterID(), event_desc);
```

Comment:
QServ isn't used anymore. We should use the modern player event system to capture this

Reply:
Sounds good.  I will update